module/zfs/spa_errlog.c

```diff
@@ -306,23 +307,6 @@ check_filesystem(spa_t *spa, uint64_t head_ds, zbookmark_err_phys_t *zep,
 	boolean_t check_snapshot = B_TRUE;
 	error = find_birth_txg(ds, zep, &latest_txg);
 
-	/*
-	 * If the filesystem is encrypted and the key is not loaded
-	 * or the encrypted filesystem is not mounted the error will be EACCES.
```

Comment:
Isn't the first case here still valid, "the filesystem is encrypted and the key is not loaded"?  It's just the second mount case which can no longer happen.  What am I missing?

Reply:
Yes, it is valid. However, even without that code block, `zpool status -v` still returns `permission denied`, as it should. That should be coming from `dsl_dataset_hold_obj_flags()` erroring out above.